### PR TITLE
【1.1.13】Workflow shell node viewing resource file, path usage error

### PR DESCRIPTION
**Pull request by v-kkhuang**
### What is the purpose of the change

Workflow shell node viewing resource file, path usage error

### Related issues/PRs

Related issues: #4585

### Brief change log

- Define the core abstraction and interfaces of the EngineConn Factory;
- Define the core abstraction and interfaces of Executor Manager.

### Checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] I have explained the need for this PR and the problem it solves
- [ ] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [Linkis mailing list](https://linkis.apache.org/community/how-to-subscribe) first)
- [ ] **If this is a code change**: I have written unit tests to fully verify the new behavior.

### CODE REVIEW

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala [hunk 1]
```diff
@@ -63,7 +64,7 @@ class CSResourceParser extends Logging {
       val bmlResourceOption =
         bmlResourceList.asScala.find(_.getDownloadedFileName.equals(resourceName))
       if (bmlResourceOption.isDefined) {
-        val replacementName = prefixName + resourceName
+        val replacementName = EngineConnConf.getWorkHome + "/tmp/" + prefixName + resourceName
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala:67]: EngineConnConf.getWorkHome  result example as ？

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala [hunk 2]
```diff
@@ -71,9 +72,9 @@ class BmlEnginePreExecuteHook extends ComputationExecutorHook with Logging {
...
               val resourceId = resource.get(GovernanceConstant.TASK_RESOURCE_ID_STR).toString
               val version = resource.get(GovernanceConstant.TASK_RESOURCE_VERSION_STR).toString
-              val fullPath =
-                if (workDir.endsWith(seperator)) pathType + workDir + fileName
-                else pathType + workDir + seperator + fileName
+              val fullPath = fileName
+//                if (workDir.endsWith(seperator)) pathType + workDir + fileName
+//                else pathType + workDir + seperator + fileName
               val response = Utils.tryCatch {
                 bmlClient.downloadShareResource(processUser, resourceId, version, fullPath, true)
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala:79]: why remove prefix path  for download?

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala (hunk 1 above)
Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala:67]: image
Enginepath/tmp/

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala (hunk 2 above)
Comment (v-kkhuang) [reply linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala:79]: The replacementName already has the full address, so we need to cancel the splicing here

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala (hunk 1 above)
Comment (casionone) [reply linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/cs/CSResourceParser.scala:67]: batter only modify at download  bml

File: linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala [hunk 3]
```diff
@@ -71,9 +72,9 @@ class BmlEnginePreExecuteHook extends ComputationExecutorHook with Logging {
               val fileName = resource.get(GovernanceConstant.TASK_RESOURCE_FILE_NAME_STR).toString
               val resourceId = resource.get(GovernanceConstant.TASK_RESOURCE_ID_STR).toString
               val version = resource.get(GovernanceConstant.TASK_RESOURCE_VERSION_STR).toString
-              val fullPath =
-                if (workDir.endsWith(seperator)) pathType + workDir + fileName
-                else pathType + workDir + seperator + fileName
+              val fullPath = fileName
+//                if (workDir.endsWith(seperator)) pathType + workDir + fileName
+//                else pathType + workDir + seperator + fileName
```
Comment (casionone) [inline linkis-computation-governance/linkis-engineconn/linkis-computation-engineconn/src/main/scala/org/apache/linkis/engineconn/computation/executor/bml/BmlEnginePreExecuteHook.scala:77]: just only need to set fullpath

**Pull request merged**